application-load-balancer/lambda_function/secret_rotator.py

```diff
@@ -0,0 +1,450 @@
+
+        Args:
+            distro_id (str): The ID of the CloudFront distribution.
+            header_value (str): The header value to set for the custom header.
+        """
+        client = self.get_cloudfront_session()
+
+        if not self.is_distribution_deployed(distro_id):
+            logger.error("Distribution Id, %s status is not Deployed." % distro_id)
+            raise ValueError(f"Distribution Id, {distro_id} status is not Deployed.")
+
+        dist_config = self.get_cfdistro_config(distro_id)
+
+        updated = self.update_custom_headers(dist_config, header_value)
+
+        if not updated:
+            logger.error("No custom header, %s found in distribution Id, %s." % (self.header_name, distro_id))
+            raise ValueError(f"No custom header found in distribution Id, {distro_id}.")
+
+        # Update the distribution
+        return self.apply_distribution_update(client, distro_id, dist_config)
+
+    def is_distribution_deployed(self, distro_id: str) -> bool:
+        """
+        Checks if the CloudFront distribution is deployed.
+
+        """
+        dist_status = self.get_cfdistro(distro_id)
+        return 'Deployed' in dist_status['Distribution']['Status']
+
+    def update_custom_headers(self, dist_config: Dict, header_value: str) -> bool:
+        """
+        Updates custom headers in the distribution config.
+        """
+        header_count = 0
+        for k in dist_config['DistributionConfig']['Origins']['Items']:
+            if k['CustomHeaders']['Quantity'] > 0:
+                for h in k['CustomHeaders']['Items']:
+                    if self.header_name in h['HeaderName']:
+                        logger.info(f"Update custom header, {h['HeaderName']} for origin: {k['Id']}.")
+                        header_count += 1
+                        h['HeaderValue'] = header_value
+                    else:
+                        logger.info(f"Ignore custom header, {h['HeaderName']} for origin: {k['Id']}.")
+            else:
+                logger.info(f"No custom headers found in origin: {k['Id']}.")
+        
+        return header_count > 0
+            
+    def apply_distribution_update(self, client, distro_id: str, dist_config: Dict) -> Dict:
+        """
+        Applies the distribution update to CloudFront.
+        """
+        try:
+            response = client.update_distribution(
+                Id=distro_id,
+                IfMatch=dist_config['ResponseMetadata']['HTTPHeaders']['etag'],
+                DistributionConfig=dist_config['DistributionConfig']
+            )
+            
+            status_code = response['ResponseMetadata']['HTTPStatusCode']
+            
+            if status_code == 200:
+                logger.info("CloudFront distribution %s updated successfully", distro_id)
+            else:
+                logger.warning("Failed to update CloudFront distribution %s. Status code: %d", distro_id, status_code)
```

Comment:
should this not result in an exception?  Or is it ok to continue?

Reply:
Agreed. Added RunTime exception and raised errors should they occur